### PR TITLE
Change middleware to listener

### PR DESCRIPTION
**Pull request by GabrielDeveloper**

### CODE REVIEW

**Comment by HDVinnie** [comment]
@GabrielDeveloper @marcoaraujojunior any specific reason you guys went to a listener over middleware. Yes the middleware ran on every request(more load) but made the online stats alot more accurate. With the listener(less load) there is many issues such as users that use the Laravel remember_me function. They will not show unless they logout manually and then back in as the remember_me lasts along time. So they could be active on site for weeks and never show in online stats.

Theres also the issue of it just not being as "realtime" for lack of a better word. Basing it off session lifetime is bad. Given that the user's session is a timeout. Every time it's accessed, the time to expiry is reset to the initial value.
In other words, if you set a cache to the session timeout, say it's 120 minutes, the session may not timeout for days, provided the user keeps accessing it before 120 minutes elapses at any given time.

**Comment by GabrielDeveloper** [comment]
@HDVinnie first of all thank your interesting, the main change of this PR its because some packages like Passport or JWT doens't use the "Auth::logout()" to "revoke" the access and we had some problems because even after the user made logoff , its been put on online user list again. The reason is because the user was not out of Laravel Auth session, and the middleware placed him again.

I already made the PR to implement the events on this packages [Passport](https://github.com/laravel/passport/pull/1018) and [JWT](https://github.com/tymondesigns/jwt-auth/pull/1813). The Passaport PR unhappily was denied because they think this doesn't feel like a good use case for that.

The JWT PR is open and if they accept I'll implement the listeners of this events to make compatible with our the package.

About the "renew cache" and session life, it is been analize other way to use and set, if you has other solution please share with us.

**Comment by MoeCasts** [comment]
@GabrielDeveloper Is possible to use websockets to get users online status instead of listener?
